### PR TITLE
now serialzy supports saving serialized model and its meta into separ…

### PR DESCRIPTION
**Pull request by zhvkgj**
…ate files

### CODE REVIEW

File: serialzy/serializers/base_model.py [hunk 1]
```diff
@@ -118,3 +123,50 @@ def __check_module_fits_serializer(self, typ: Type) -> bool:
...
+        :param dest_dir: directory with files with serialized model and meta
+        :param model_file_name: name of file with serialized model
+        :param typ: type of the resulting object, fetched from the meta if None
+        :return:
+        """
+        dest_dir_path = Path(dest_dir)
+
+        # read schema header
+        meta_file_path = dest_dir_path / ModelBaseSerializer.META_FILE_NAME
+        with open(meta_file_path, 'rb') as source:
```
Comment (vhaldemar) [inline serialzy/serializers/base_model.py:166]: If you using pathlib, you should use it fully:
`with meta_file_path.open('rb') as source:`

**Review by imakunin — APPROVED** [review]
LGTM

File: serialzy/serializers/base_model.py [hunk 2]
```diff
@@ -68,6 +71,8 @@ def deserialize_from_dir(source: BinaryIO, read_from_dir) -> Any:
 
 
 class ModelBaseSerializer(DefaultSchemaSerializerByReference, ABC):
+    META_FILE_NAME = 'meta.json'
```
Comment (vhaldemar) [inline serialzy/serializers/base_model.py:74]: I'm not a fan of CAPS at class-level constants.
Maybe in the future or in descendants this field would be a property.

File: serialzy/serializers/base_model.py [hunk 3]
```diff
@@ -118,3 +123,50 @@ def __check_module_fits_serializer(self, typ: Type) -> bool:
...
+        """
+        :param obj: object to serialize into bytes
+        :param dest_dir: files with serialized model and meta are written into destination directory
+        :param model_file_name: serialized model's file name
+        :return: None
+        """
+        dest_dir_path = Path(dest_dir)
+
+        if not dest_dir_path.is_dir():
+            raise NotADirectoryError
```
Comment (vhaldemar) [inline serialzy/serializers/base_model.py:137]: Here and downwards you a missing useful message like `f'given path {dest_dir_path} is not a dir'`

File: tests/rich_env/serializers/test_base_model.py [hunk 4]
```diff
@@ -4,27 +4,36 @@
...
+
+    def base_test(self, model: Any, expected_serializer):
+        serializer = self.registry.find_serializer_by_type(type(model))
+        self._assert_model_serializer(serializer, expected_serializer)
+        return serialize_and_deserialize(serializer, model)
+
+    def base_test_with_meta(self, model: Any, expected_serializer):
+        serializer = self.registry.find_serializer_by_type(type(model))
+        self._assert_model_serializer(serializer, expected_serializer)
+        assert isinstance(serializer, ModelBaseSerializer)
```
Comment (vhaldemar) [inline tests/rich_env/serializers/test_base_model.py:35]: `self.assertIsInstance`

File: serialzy/serializers/base_model.py [hunk 5]
```diff
@@ -118,3 +123,50 @@ def __check_module_fits_serializer(self, typ: Type) -> bool:
...
+        meta_file_path = dest_dir_path / ModelBaseSerializer.META_FILE_NAME
+        with open(meta_file_path, 'wb') as dest:
+            self._write_schema(typ, dest)
+
+        # write serialized model
+        model_file_path = dest_dir_path / model_file_name
+        with open(model_file_path, 'wb') as dest:
+            self._serialize(obj, dest)
+
+    def deserialize_with_meta(self, dest_dir: str, model_file_name: str, typ: Optional[Type] = None) -> Any:
```
Comment (vhaldemar) [inline serialzy/serializers/base_model.py:155]: dest_dir -> source_dir

File: serialzy/version/version [hunk 6]
```diff
@@ -1 +1 @@
-1.1.2
+1.2.2
```
Comment (imakunin) [inline serialzy/version/version:1]: 1.2.0 :)

Comment (zhvkgj) [reply serialzy/version/version:1]: oops

File: serialzy/serializers/base_model.py [hunk 7]
```diff
@@ -118,3 +123,50 @@ def __check_module_fits_serializer(self, typ: Type) -> bool:
                 if base_name == self.module:
                     return True
         return False
+
+    def serialize_with_meta(self, obj: Any, dest_dir: str, model_file_name: str) -> None:
```
Comment (vhaldemar) [inline serialzy/serializers/base_model.py:127]: I generally don't like the abbreviationы like `dest_dir`. `destination_directory`, `destination` and such seems a lot better.

File: tests/rich_env/serializers/test_base_model.py [hunk 8]
```diff
@@ -4,27 +4,36 @@
...
 class ModelBaseSerializerTests(TestCase):
     def initialize(self, module: str):
         self.module = module
         self.registry = DefaultSerializerRegistry()
 
-    def base_test(self, model: Any, expected_serializer):
-        serializer = self.registry.find_serializer_by_type(type(model))
-        deserialized_model = serialize_and_deserialize(serializer, model)
-
+    def _assert_model_serializer(self, serializer, expected_serializer):
```
Comment (vhaldemar) [inline tests/rich_env/serializers/test_base_model.py:20]: `-> None`  to turn on mypy here and downwards

**Review by vhaldemar — CHANGES_REQUESTED** [review]

File: tests/rich_env/serializers/test_base_model.py (hunk 8 above)
Comment (zhvkgj) [reply tests/rich_env/serializers/test_base_model.py:20]: common style of this file forces absence of returning type declaration

File: serialzy/serializers/base_model.py [hunk 9]
```diff
@@ -1,13 +1,16 @@
+import abc
```
[the remark this answers is not in the document]
Comment (zhvkgj) [reply serialzy/serializers/base_model.py:1]: i missed it, attentive comment

**Review by vhaldemar — APPROVED** [review]

**Pull request merged**